enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,273 @@
+* Controller for registering CRs with an API for listing, extracting and downloading plugins
+  * An API that generates an index that `krew` can consume via its custom index feature
+  * The index and binaries will served from a single route and service within the cluster
+* The `krew` client package will be vendored into `oc` for usage as `oc krew install <plugin>`
+
+### Non-Goals
+
+* The Controller will not build or package plugins
+* No recommending/limiting which plugins are served by the API
+* `krew` will not create or update the Plugin CRs, those will be managed by cluster-admins
+
+## Proposal
+
+Each component wishing to provide customers with their plugins will build and publish images via a trusted image registry
+and create a Plugin CR to provide an image name and the file path within that image for the binaries.
+Clients (i.e. `krew`) will read the index and download binaries from the Controller. The Controller is responsible for building the index from CRs, for pulling the images,
+extracting the binaries, and serving them to clients.
+
+* `krew` and `krew` plugins are upstream projects that Kubernetes users are already familiar with
+* A `krew`-compatible custom index can provide available plugins for a cluster in disconnected environments
+* The index will be served by a Controller with its contents managed by cluster-admins via CRs
+* The binaries will also be served by the Controller that will pull images from a trusted image registry and extract the binaries
+* With `krew index add https://someother-third-party-index` we won't limit users from adding their own index with whatever plugins they want
+* The controller could be optional, since connected environments can use the default `krew` index out-of-the box
+  * We feel as though the controller should be installed by default so that we can replace the existing temporary solution with something more robust
+  * Pros for being installed by default:
+    * Available day one for all clusters
+    * All binaries can be served using single unified mechanism
+    * No user setup required to use in disconnected environments
+  * Cons:
+    * Additional controller running by default
+  * Pros for being optionally installed:
+    * Less resource utilization
+  * Cons:
+    * Another operator to manage and mirror for disconnected environments
+    * Separate mechanism for built-in and third-party binaries
+    * Missing plugin CRs for optional elements installed before controller is installed
+* As `krew` itself is a `kubectl` plugin, it can be invoked using either using `kubectl krew` or `oc krew`\
+* `krew` functionality baked into `oc` by default
+
+Existing methods of downloading binaries (i.e. the console) will not be affected by this proposal. For the initial implementation, supported plugins will create Plugin CRs. The plugins will be downloaded and installed using `krew`.
```

Comment:
Given we want to go the default route, the existing mechanism will be replaced with this new one, but it should be transparent for the consumers, except for the CR change.

Reply:
for it to be transparent to consumers doesn't that mean the console needs to learn to process the new CR?

is the console team lined up to participate in this change?  or if not what does the transition look like? (will we have both the built-in mechanism and this new one, at least temporarily during the transition period?)